lnwallet/btcwallet/btcwallet.go

```diff
@@ -789,28 +792,48 @@ func (b *BtcWallet) ListAddresses(name string) ([]string, error) {
 			return nil, err
 		}
 		for _, account := range accounts.Accounts {
-			account := account
-			res = append(res, account.AccountProperties.AccountNumber)
-		}
+			accountProperties := account.AccountProperties
+			res[accountProperties.AccountName] = accountDetails{accountName: accountProperties.AccountName, accountNumber: accountProperties.AccountNumber}
+			}
 	}
 
 	type addressProperty struct {
 		address  string
 		internal bool
+		balance  btcutil.Amount
```

Comment:
The `addressProperty` should probably be an exported type that this method returns (instead of just `[]string`).